awsx/ecr/image.ts

```diff
@@ -62,37 +62,52 @@ export function computeImageFromAsset(
       throw new Error("Invalid credentials");
     }
     return {
-      registry: ecrCredentials.proxyEndpoint,
+      address: ecrCredentials.proxyEndpoint,
       username: username,
       password: password,
     };
   });
 
+  let cacheFrom: docker.types.input.CacheFromArgs[] = [];
+  if (dockerInputs.cacheFrom !== undefined) {
+    cacheFrom = dockerInputs.cacheFrom.map((c) => {
+      return {
+        registry: {
+          ref: c,
+        },
+      };
+    });
+  }
+  // Use an inline cache by default.
+  if (cacheFrom.length === 0) {
+    cacheFrom.push({ registry: { ref: canonicalImageName } });
+  }
+
+  let context = ".";
+  if (dockerInputs.context !== undefined) {
+    context = dockerInputs.context;
+  }
+
   const dockerImageArgs: docker.ImageArgs = {
```

Comment:
Are there any new input properties we should expose as well?

Reply:
Probably! There's a lot more we can do now, e.g. multiple tags, multiple platforms, more refined caching behavior, etc.

I didn't want to add any of those right now because (a) these are opinionated APIs and opinions take more time to get concensus on, and (b) it's not clear how much (if any) we want to continue investing in this wrapper versus giving users more composable building blocks (like your example of registry auth helpers). The latter is my preference, but it's not my call to make.